src/error.rs

```diff
@@ -50,3 +50,20 @@ impl Display for HidError {
 }
 
 impl Error for HidError {}
+
+impl From<std::io::Error> for HidError {
+    fn from(e: std::io::Error) -> Self {
+        Self::HidApiError {
+            message: format!("{e}"),
+        }
+    }
+}
+
+#[cfg(linux_native)]
+impl From<nix::errno::Errno> for HidError {
+    fn from(e: nix::errno::Errno) -> Self {
```

Comment:
This feels like a different situation from `std::io::Error` as it would add a variant that only exists with some options, which seems like it's going to make it a pain for users of the library to deal with. But if `std::io::Error` does become a variant, then maybe we can make a decent effort at translating it to that.

Reply:
Given that any nix Errno is going to be convertible into a `std::io::Errror` and they provide a impl for that, I've made it so nix errors convert into the `IoError` variant which lets us avoid an extra equivalent variant.

Reply:
Good idea, lets do it like that